### PR TITLE
Feature - 5496 - Admin teams index page

### LINKED ISSUE #5496: ✨ (Admin) List Teams page
## ✨ Feature

This page shows all the teams on the platform in a table, is the highest level page for all team related content. May want to work on this alongside #5511.

## 🕵️ Details

This is a new page and should be reachable through the main navigation.

image

In addition to having a table that lists all teams on the platform:
- it has a "Create new team" button that leads to #5497
- you can click on Edit team link #5499
- you can click on Delete team link #5500
- you can click on a team name to view that team #5501
- you can click on a view members to view that teams members' subpage #5502

## 🎨 Design File

Link to figma

## 📸 Screenshot

image

## 🙋‍♀️ Proposed Implementation

Use the Table component, not the ApiManagedTable, because there will be few rows. Use the default search functionality, but ignore the Filter button.

Though it may not be implemented, assume the data is coming from a `teams: Team[]` query in our schema, with the Team definition looking like:
```
type Team {
  name: String
  displayName: LocalizedString
  description: LocalizedString
  departments: Department[]
  contactEmail: Email
}
```  
You may want to mock this response in StoryBook to work on this story.

The New Team, Edit Team, View Team, and View Members buttons are links to the following pages, whether the page exists or not:
- new team: `/admin/teams/create`
- view team: `/admin/teams/{name-slug}`
- edit team: `/admin/teams/{name-slug}/edit`
- members: `/admin/teams/{name-slug}/members`

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] The table lists all Teams returned by the `teams` graphQL query 
- [ ] This page is hosted at `/admin/teams`
- [ ] Links direct you to the correct urls, even if they result in 404s for now
- [ ] This page is only accessible with the Admin role

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #5496 

## 👋 Introduction

This adds the index page for teams to the admin portal.

## 🕵️ Details

This is only the index page, other pages will be added in different PRs but, it creates the paths and routes ready for use.

## 📋 TO DO

- [x] Wait for #5511 
- [x] Replace mocks with API requests
- [x] Add translations

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Build the project `npm run build`
2. Login to admin
3. Confirm the "Teams" nav item exists on the sidebar nav
4. Navigate to `/admin/teams`
5. Confirm page loads and table functions as expected
    - Can sort each columns
    - Search functions
    - Can hide/remove colums

## 📸 Screenshot
localhost_8000_en_admin_teams_sortBy=%255B%257B%2522id%2522%253A%2522Email%2522%252C%2522desc%2522%253Atrue%257D%255D hiddenColumns=

### CODE REVIEW

**Comment by vd1992** [comment]
oh, you got merge conflicts 📬

File: frontend/common/src/fakeData/fakeTeams.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { faker } from "@faker-js/faker";
+// import {Team}from "../api/generated";
+
+// TO DO: Remove after #5548
```
Comment (vd1992) [inline frontend/common/src/fakeData/fakeTeams.ts:4]: I believe this can be cleaned up now?

**Comment by vd1992** [comment]
given that a team can have multiple departments associated with it, the department column may need some further consideration
it currently just pinches the first department in the array, so it isn't broken but a bit odd

image

File: apps/web/src/pages/Teams/IndexTeamPage/components/TeamTable/TeamTable.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import fakeTeams from "@common/fakeData/fakeTeams";
+
+import { TeamTable } from "./TeamTable";
+
+const mockTeams = fakeTeams();
+
+export default {
+  component: TeamTable,
+  title: "Tables/Department Table",
+} as ComponentMeta<typeof TeamTable>;
```
Comment (vd1992) [inline apps/web/src/pages/Teams/IndexTeamPage/components/TeamTable/TeamTable.stories.tsx:13]: think you'll need a different title for the story here
same as this one

https://github.com/GCTC-NTGC/gc-digital-talent/blob/d9a45927cb53323313c5366b8feb200421919cb9/apps/web/src/pages/Departments/components/DepartmentTable.stories.tsx#L10-L13

it's making weirdness in the chromatic diff

image

**Review by vd1992 — CHANGES_REQUESTED** [review]
sorting appears to work
search works for departments but not team or email columns

https://recordit.co/SIWDJvjnm9

File: frontend/common/src/fakeData/fakeTeams.ts (hunk 1 above)
Comment (esizer) [reply frontend/common/src/fakeData/fakeTeams.ts]: Good catch! Removed in [adjust fake teams, remove to do](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5555/commits/9f367b98d519531b361024289ab9f29d32e600d2)

File: apps/web/src/pages/Teams/IndexTeamPage/components/TeamTable/TeamTable.stories.tsx (hunk 2 above)
Comment (esizer) [reply apps/web/src/pages/Teams/IndexTeamPage/components/TeamTable/TeamTable.stories.tsx]: heh, missed copy + past 😅 

Fixed in [fix copy paste typo](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5555/commits/49fec9aad78ce61760362811de2ff0b8a168ef67)

**Comment by esizer** [comment]
> given that a team can have multiple departments associated with it, the department column may need some further consideration
> it currently just pinches the first department in the array, so it isn't broken but a bit odd

Yeah, I wasn't sure at the time and totally forgot to update this once I knew it could have multiple departments. Thanks for catching that 👍 Fixed in [adjust teams to support multiple departments](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5555/commits/b1fed53e19a200b9978436c5d0883a305995d376)

> sorting appears to work
> search works for departments but not team or email columns

lol, reading the docs helps 😅 Fixed in [fix table search](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5555/commits/c518537faf937534f657f900d6d365dc0c391979)

**Review by vd1992 — APPROVED** [review]
great!

**Pull request merged**